src/components/Basket/OrderSummary/OrderSummary.tsx

```diff
@@ -0,0 +1,44 @@
+import { H3 } from "../../common/headings/H3";
+import { TfiLayoutLineSolid } from "react-icons/tfi";
+import OrderField from "./OrderField/OrderField";
+import OrderPromoCode from "./OrderPromoCode/OrderPromoCode";
+
+import "./OrderSummaryStyles.scss";
+import { useCart } from "../../context/useCart";
+
+type priceType = "total" | "discount" | "subtotal";
+
+export default function OrderSummary() {
+  const { cart } = useCart();
+
+  const handlePrice = (priceType: priceType) => {
+    const totalPriceCents = cart?.totalPrice?.centAmount || 0;
+    const totalPriceFraction = cart?.totalPrice?.fractionDigits || 0;
+    const totalPrice = totalPriceCents / 100;
+
+    const discountedPriceCents = cart?.discountOnTotalPrice?.discountedAmount?.centAmount || 0;
+    const discountedPriceFraction = cart?.discountOnTotalPrice?.discountedAmount?.fractionDigits || 0;
+    const discountedPrice = discountedPriceCents / 100;
+
+    if (priceType === "total") {
+      return "€" + totalPrice.toFixed(totalPriceFraction);
+    } else if (priceType === "discount") {
+      return "€" + discountedPrice.toFixed(discountedPriceFraction);
+    } else {
+      return "€" + (discountedPrice + totalPrice).toFixed(totalPriceFraction);
```

Comment:
too many extra calculations for every priceType... basically, you calculating everything 3 times since you are calling this function 3 times... why not to calculate them all only once and keep them in variables? or you can also use one useMemo for all priceType (eg as an object)